src/data/PredictionManager.ts

```diff
@@ -261,6 +261,13 @@ function generated1DSpectrum(params: {
     nbPoints,
     lineWidth,
   });
+
+  const first = x[0] ?? 0;
```

Comment:
@targos What is the best way to solve the undefined issue in this code ?

Reply:
The problem with TS. Should we make a check that length is >0 and throw otherwise ? Should we simply do `as number` ?

Reply:
It depends on the situation:
- If the array may be empty, the current code seems good
- If you expect it to never be empty, either throw or use `as number`